python/ray/serve/tests/test_gradio.py

```diff
@@ -0,0 +1,71 @@
+import os
+import sys
+import pytest
+import requests
+
+import ray
+from ray import serve
+from ray.serve.gradio import GradioIngress
+import gradio as gr
+
+
+@pytest.fixture
+def serve_start_shutdown():
+    ray.init()
+    serve.start()
+    yield
+    serve.shutdown()
+    ray.shutdown()
+
+
+def test_gradio_ingress_correctness(serve_start_shutdown):
+    """
+    Ensure a Gradio app deployed to a cluster through GradioIngress still
+    produces the correct output
+    """
+
+    def greet(name):
+        return f"Good morning {name}!"
+
+    io = gr.Interface(fn=greet, inputs="text", outputs="text")
+    GradioServer = serve.deployment(GradioIngress)
```

Comment:
Can we make `GradioIngress` a deployment by default so users don't need to do the `serve.deployment()` call? I don't see a downside except that it would preclude the ability to inherit from the class (we could have a separately importable class and deployment though).

Reply:
Yes that's what I was thinking.